src/Services/GameDataService.cs

```diff
@@ -33,39 +36,61 @@ public class GameDataService {
         ctx.SaveChanges();
         return true;
     }
-
-    public GameDataSummary GetGameData(Viking viking, int gameId, bool isMultiplayer, int difficulty, int gameLevel, string key, int count, bool AscendingOrder, bool buddyFilter, string apiKey, DateTime? startDate = null, DateTime? endDate = null) {
-        // TODO: Buddy filter
-        List<GameDataResponse> selectedData;
-        IQueryable<Model.GameData> query = ctx.GameData.Where(x => x.GameId == gameId && x.IsMultiplayer == false && x.Difficulty == difficulty && x.GameLevel == gameLevel);
-
-        if (startDate != null && endDate != null)
-            query = query.Where(x => x.DatePlayed >= startDate.Value.ToUniversalTime() && x.DatePlayed <= endDate.Value.AddMinutes(2).ToUniversalTime());
-
-        var query2 = query.SelectMany(e => e.GameDataPairs)
+	
+	List<GameDataResponse> GameDataResponseToList(IQueryable<Model.GameData> query, string key, int count, bool AscendingOrder, string apiKey) {
+		var query2 = query.SelectMany(e => e.GameDataPairs)
             .Where(x => x.Name == key);
+			
+		if (AscendingOrder)
+			query2 = query2.OrderBy(e => e.Value);
+		else
+			query2 = query2.OrderByDescending(e => e.Value);
 
         uint gameVersion = ClientVersion.GetVersion(apiKey);
         if (gameVersion <= ClientVersion.Max_OldJS) {
             // use DisplayName instead of Name
             if (AscendingOrder)
```

Comment:
if you previously defined `query2` dependent on the sorting order, you can remove this and the next `if (AscendingOrder)` (they do nothing - both branches are the same)